src/vs/vscode.d.ts

```diff
@@ -4967,6 +4967,15 @@ declare module 'vscode' {
 		 */
 		label: string;
 
+		/**
+		 * Optional id for the tree item that has to be unique across tree. If not provided, an unique id is generated using the tree item's label.
+		 *
+		 * This is used by the view to preserve the selection and expansion state of the tree item.
+		 *
+		 * *Note:* With generated id, tree item's view state will be reset when label is changed.
```

Comment:
Not sure what that mean? With a *re*-generated id state will reset?

Reply:
Yes,  if the label has changed, new `id` (different from old) is generated and as a result previous state is lost.

Reply:
Ok, how about merging that with sentence about how an id will be generated. 

```
Optional id for the tree item that has to be unique across tree. The id is used to preserve the selection and expansion state of the tree item.

If not provided, an id is generated using the tree item's label. **Note** that when labels change, ids will change and that selection and expansion state cannot be kept stable anymore. 
```

Reply:
Looks better. Updated.